backends/qualcomm/scripts/download_qnn_sdk.py

```diff
@@ -0,0 +1,247 @@
+# Add these imports for additional logging
+import ctypes
+import os
+import pathlib
+import platform
+import shutil
+import tarfile
+import tempfile
+import urllib.request
+import zipfile
+from typing import Optional
+
+from tqdm import tqdm
+
+PKG_ROOT = pathlib.Path(__file__).parent.parent
+SDK_DIR = PKG_ROOT / "sdk" / "qnn"
+
+
+def is_linux_x86() -> bool:
+    """
+    Check if the current platform is Linux x86_64.
+
+    Returns:
+        bool: True if the system is Linux x86_64, False otherwise.
+    """
+    return platform.system().lower() == "linux" and platform.machine().lower() in (
+        "x86_64",
+        "amd64",
+        "i386",
+        "i686",
+    )
+
+
+def _download_qnn_sdk() -> Optional[pathlib.Path]:
+    """
+    Download and extract the Qualcomm SDK into SDK_DIR.
+
+    Notes:
+        - Only runs on Linux x86 platforms. Skips otherwise.
+    """
+    print("Downloading Qualcomm SDK...")
+    QNN_VERSION = "2.37.0.250724"
```

Comment:
nit: maybe refactor to get/set qnn version functions? Mainly for ease of maintenance.

Reply:
Yeah I'd need to have a separate PR to remove the hardcoded numbers https://github.com/pytorch/executorch/pull/13960